crates/node/service/src/service/core.rs

```diff
@@ -0,0 +1,41 @@
+//! Core [RollupNodeService] trait.
+
+use super::{SequencerNodeService, ValidatorNodeService};
+use async_trait::async_trait;
+use tracing::info;
+
+/// The [NodeMode] enum represents the modes of operation for the [RollupNodeService].
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NodeMode {
+    /// Validator mode.
+    Validator,
+    /// Sequencer mode.
+    Sequencer,
+}
+
+/// The [RollupNodeService] trait defines the core functionality of a rollup node. It can operate
+/// in two modes: as a [ValidatorNodeService] or as a [SequencerNodeService].
+///
+/// [SequencerNodeService]: super::SequencerNodeService
+#[async_trait]
+pub trait RollupNodeService: ValidatorNodeService + SequencerNodeService {
+    /// Returns the [NodeMode] of the service.
+    fn mode(&self) -> NodeMode;
+
+    /// Starts the rollup node service, using the entry point respective to the [NodeMode].
+    ///
+    /// - [NodeMode::Validator]: [ValidatorNodeService::start]
+    /// - [NodeMode::Sequencer]: [SequencerNodeService::start]
+    async fn start(&self) -> Result<(), Self::Error> {
+        info!(target: "rollup_node", "Starting rollup node services");
+        info!(target: "rollup_node", "Chain ID: {}", self.config().l2_chain_id);
+        for hf in self.config().hardforks.to_string().lines() {
+            info!(target: "rollup_node", "{hf}");
+        }
```

Comment:
This is just to print each display line on a new log line?

Reply:
Yeah - mainly to keep everything in tracing rather than a raw write to stdout. Prints out the list of hard fork activation times configured on startup.

Reply:
Very clean